app/src/main/java/com/example/makeitso/screens/sign_up/SignUpViewModel.kt

```diff
@@ -75,42 +75,18 @@ class SignUpViewModel @Inject constructor(
             return
         }
 
-        viewModelScope.launch(showErrorExceptionHandler) {
-            val oldUserId = accountService.getUserId()
-            val createAccountTrace = Firebase.performance.newTrace(CREATE_ACCOUNT_TRACE)
-            createAccountTrace.start()
-
-            accountService.createAccount(email, password) { error ->
-                createAccountTrace.stop()
-
-                if (error == null) {
-                    linkWithEmail()
-                    updateUserId(oldUserId, openAndPopUp)
-                } else onError(error)
-            }
+        launchCatching {
+            val oldUserId = accountService.currentUserId
+            val newUser = accountService.createAccount(email, password)
+            linkWithEmail()
```

Comment:
When I try to sign up, it works but I see a snackbar saying that this account has already been linked (it seems like maybe the link method is being called twice?) - is it also happening with you?

Reply:
Hm, upon a closer look, I don't think we are supposed to "link" the just registered email account as it is associated with the email provider by definition. So linking is not required. That said though, given that we are always guaranteed to have an anonymous user, we can just remove `createAccount` and only keep linkWithEmail. That way we don't need to migrate the data upon login as the uid will stay the same, but the user will now have email credentials. wdyt?

Reply:
I verified that that is the case and what we likely want https://firebase.google.com/docs/auth/android/anonymous-auth#convert-an-anonymous-account-to-a-permanent-account

Updated the code. wdyt?

Reply:
You're right! And the code looks great 👏 
I ran on my device and everything works flawlessly!